common/cluster/metadata.go

```diff
@@ -56,8 +71,15 @@ type (
 		ClusterNameForFailoverVersion(failoverVersion int64) string
 	}
 
+	searchAttributesCache struct {
+		searchAttributes map[string]*persistencespb.IndexSearchAttributes
+		dbVersion        int64
+		lastRefresh      time.Time
+	}
```

Comment:
search attribute management should belong to its own struct, not within cluster metadata